### PR TITLE
Testing setup

### PR DESCRIPTION
**Pull request by debjit-bw**
Adds necessary handlers and drivers for testing with ethereum/tests and the new eels tests.

* We are skipping tests targeted at the following forks:
    - Frontier
    - Homestead
    - Byzantium
    - ByzantiumToConstantinopleAt5
    - Istanbul
    - Berlin
    - London
    - Constantinople
    - ConstantinopleFix
    - MergeEOF
    - MergeMeterInitCode
    - MergePush0
>  This DOES NOT mean that the features introduced at these forks are untested. Any feature introduced have tests targeted at all the forks after that, and we ARE testing for the fork "Merge" and later, skipping the ones before it.
* We are directly using the tests generated for Ethereum, since we don't currently have the tests generated for Gnosis. 
    * Therefore, several tests are failing, particularly 1559, withdrawals etc.
    * For most tests however, Gnosis and Mainnet work similarly.
    * 53 out of the 71 tests (combining eest and ethereum/tests) are PASSING. For now I have feature-gated the failing tests.
    * 18 tests are inside `failing-tests` feature (25%).

This PR is targeted mostly at setting up the code needed to run the EELS or ethereum tests. The tests themselves will be kept updated as and when Gnosis tests are available. The tests currently passing is just to prove that we aren't doing anything wildly wrong.

### CODE REVIEW

File: src/testing/cases/blockchain_test.rs [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+
+        // Skip test where basefee/accesslist/difficulty is present but it shouldn't be supported in
+        // London/Berlin/TheMerge. https://github.com/ethereum/tests/blob/5b7e1ab3ffaf026d99d20b17bb30f533a2c80c8b/GeneralStateTests/stExample/eip1559.json#L130
+        // It is expected to not execute these tests.
+        | "accessListExample.json"
+        | "basefeeExample.json"
+        | "eip1559.json"
+        | "mergeTest.json"
+
+        // These tests are passing, but they take a lot of time to execute so we are going to skip them.
```
Comment (dapplion) [inline src/testing/cases/blockchain_test.rs:275]: How long for curiosity?

File: src/testing/cases/blockchain_test.rs [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+        | "shiftCombinations.json"
+    )
+    || matches!(
+        test_folder,
+        // These tests are passing, but they take a lot of time to execute
+        | "stTimeConsuming"
+    )
+    // Ignore outdated EOF tests that haven't been updated for Cancun yet.
+    || path_contains(path_str, &["EIPTests", "stEOF"])
+}
```
Comment (dapplion) [inline src/testing/cases/blockchain_test.rs:290]: Where do you explicitly ignore withdrawals and eip1559 tests?

File: src/testing/tests/tests.rs [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        general_state_test!(st_ext_codehash, stExtCodeHash);
+        general_state_test!(st_quadratic_complexity, stQuadraticComplexityTest);
+        general_state_test!(st_recursive_create, stRecursiveCreate);
+        general_state_test!(st_revert, stRevertTest);
+        general_state_test!(st_special, stSpecialTest);
+        general_state_test!(st_sstore, stSStoreTest);
+        general_state_test!(st_stack, stStackTests);
+        general_state_test!(st_static_call, stStaticCall); // passing, but conflicts with rewards contract
+        general_state_test!(st_transaction, stTransactionTest);
+        general_state_test!(vm_tests, VMTests);
```
Comment (dapplion) [inline src/testing/tests/tests.rs:135]: All of these should fail? Why?

**Review by dapplion — APPROVED** [review]
Awesome!! Love this addition, good progress

**Comment by dapplion** [comment]
I'll merge as is, we can work on passing more tests in another branch

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -79,6 +79,10 @@ jobs:
         run: cargo check
       - name: Test
         run: cargo test -- --test-threads=1 --nocapture
+      - name: Downloading ethereum/tests
+        run: git clone https://github.com/ethereum/tests ethereum-tests
+      - name: Downloading EELS fixtures released at Cancun
+        run: curl -LO https://github.com/ethereum/execution-spec-tests/releases/download/v2.1.1/fixtures.tar.gz && tar -xzf fixtures.tar.gz
```
Comment (dapplion) [inline .github/workflows/test.yml]: Was going to suggest caching, but this is very fast already

File: src/testing/cases/blockchain_test.rs (hunk 1 above)
Comment (debjit-bw) [reply src/testing/cases/blockchain_test.rs:275]: as long as it's running now (10 mins)
we don't exclude any tests now (not the `false &&`)

File: src/testing/tests/tests.rs (hunk 3 above)
Comment (debjit-bw) [reply src/testing/tests/tests.rs:135]: apart from eip 1559, withdrawals, tests rn fail for 2 main reasons:
- wrong impl on our end
- the test "should fail" and it is faililng, but the exact error string in reth (ExecutionStageError.MaxGasReached) is different than in exec specs (MAX_GAS_REACHED, for eg)

File: .github/workflows/test.yml (hunk 4 above)
Comment (debjit-bw) [reply .github/workflows/test.yml]: the `curl -LO` is done within a sec, but git clone does take some time tbh

**Pull request merged**